packages/core/src/tracing/dynamicSamplingContext.ts

```diff
@@ -75,39 +79,54 @@ export function getDynamicSamplingContextFromSpan(span: Span): Readonly<Partial<
   }
 
   const rootSpan = getRootSpan(span);
+  const rootSpanJson = spanToJSON(rootSpan);
+  const rootSpanAttributes = rootSpanJson.data;
+  const traceState = rootSpan.spanContext().traceState;
+
+  // The root span sample rate should always be applied to the DSC, even if the DSC is frozen.
+  // This is so that the downstream traces/services can use parentSampleRate in their `tracesSampler` to make consistent sampling decisions across the entire trace.
+  const traceSampleRateOverride =
+    traceState?.get('sentry.sample_rate_override') ??
+    rootSpanAttributes[SEMANTIC_ATTRIBUTE_SENTRY_OVERRIDE_TRACE_SAMPLE_RATE];
+  function applyTraceSampleRateOverrideToDsc(dsc: Partial<DynamicSamplingContext>): Partial<DynamicSamplingContext> {
+    if (typeof traceSampleRateOverride === 'number' || typeof traceSampleRateOverride === 'string') {
+      dsc.sample_rate = `${traceSampleRateOverride}`;
+    }
+    return dsc;
+  }
 
   // For core implementation, we freeze the DSC onto the span as a non-enumerable property
   const frozenDsc = (rootSpan as SpanWithMaybeDsc)[FROZEN_DSC_FIELD];
   if (frozenDsc) {
-    return frozenDsc;
+    return applyTraceSampleRateOverrideToDsc(frozenDsc);
   }
 
   // For OpenTelemetry, we freeze the DSC on the trace state
-  const traceState = rootSpan.spanContext().traceState;
   const traceStateDsc = traceState?.get('sentry.dsc');
 
   // If the span has a DSC, we want it to take precedence
   const dscOnTraceState = traceStateDsc && baggageHeaderToDynamicSamplingContext(traceStateDsc);
 
   if (dscOnTraceState) {
-    return dscOnTraceState;
+    return applyTraceSampleRateOverrideToDsc(dscOnTraceState);
```

Comment:
why is this not already on the frozen dsc from the trace state here?

Reply:
Because for example when the sample rate returned from a tracesSampler is different than on the incoming DSC. The sample rate returned from the tracesSampler wins.